substrate/zombienet-sdk/tests/staking_next/tests.rs

```diff
@@ -0,0 +1,380 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+use subxt::{
+	ext::scale_value::{Composite, ValueDef},
+	OnlineClient, PolkadotConfig,
+};
+
+use crate::staking_next::{activate_ah_client, build_network_config, set_validator_count};
+
+// Events generated by rc_client. Event here doesn't necessary mean 'event in the substrate
+// context'. It can be anything representing a point of interest for the tests.
+enum RcClientEvent {
```

Comment:
I think subxt has a cli which allows you to generate a metadata from the wasm file, and within that you have access to the subxt-compliant version of all of the runtime types, including the concrete event types. So you could, instead of this, assert that some events are emitted.

Reply:
That's true but in this case this event is not a runtime event.
It's an 'event' such as 'something happened in the test which we want to handle and probably change state. In this test there is 100% match between runtime event in test event but in general this might not be the case.

I should pick a better name since this causes confusion.